src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ManagedGroupRoutes.scala

```diff
@@ -1,59 +1,131 @@
 
 /**
   * Created by gpolumbo on 2/20/2018.
   */
-trait ManagedGroupRoutes extends UserInfoDirectives with SecurityDirectives {
+trait ManagedGroupRoutes extends UserInfoDirectives with SecurityDirectives with SamModelDirectives {
   implicit val executionContext: ExecutionContext
 
   val managedGroupService: ManagedGroupService
 
   def groupRoutes: server.Route = requireUserInfo { userInfo =>
-    path("group" / Segment) { groupId =>
-      get {
-        handleGetGroup(groupId)
-      } ~
-      post {
-        handlePostGroup(groupId, userInfo)
+    pathPrefix("group" / Segment) { groupId =>
+      val managedGroup = Resource(ManagedGroupService.managedGroupTypeName, ResourceId(groupId))
+
+      pathEndOrSingleSlash {
+        get {
+          handleGetGroup(managedGroup)
+        } ~
+        post {
+          handlePostGroup(managedGroup, userInfo)
+        } ~
+        delete {
+          handleDeleteGroup(managedGroup, userInfo)
+        }
       } ~
-      delete {
-        handleDeleteGroup(groupId, userInfo)
+      pathPrefix(Segment) { policyName =>
+        val accessPolicyName = parsePolicyName(policyName)
+
+        pathEndOrSingleSlash {
+          get {
+            handleListEmails(managedGroup, accessPolicyName, userInfo)
+          } ~
+          put {
+            handleOverwriteEmails(managedGroup, accessPolicyName, userInfo)
+          }
+        } ~
+        pathPrefix(Segment) { email =>
+          pathEndOrSingleSlash {
+            put {
+              handleAddEmailToPolicy(managedGroup, accessPolicyName, email, userInfo)
+            } ~
+            delete {
+              handleDeleteEmailFromPolicy(managedGroup, accessPolicyName, email, userInfo)
+            }
+          }
+        }
       }
     }
   }
 
-  private def handleGetGroup(groupId: String) = {
+  private def parsePolicyName(policyName: String): ManagedGroupPolicyName = {
+    if (policyName == "members") {
+      ManagedGroupService.memberPolicyName
+    } else if (policyName == "admins") {
+      ManagedGroupService.adminPolicyName
+    } else {
+      throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Policy name for managed groups must be one of: [\"admins\", \"members\"]"))
+    }
+  }
+
+  private def handleGetGroup(managedGroup: Resource): Route = {
     complete (
-      managedGroupService.loadManagedGroup(ResourceId(groupId)).map {
+      managedGroupService.loadManagedGroup(managedGroup.resourceId).map {
         case Some(response) => StatusCodes.OK -> response
         case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "group not found"))
       }
     )
   }
 
-  private def handlePostGroup(groupId: String, userInfo: UserInfo) = {
-    complete(managedGroupService.createManagedGroup(ResourceId(groupId), userInfo).map(_ => StatusCodes.Created))
+  private def handlePostGroup(managedGroup: Resource, userInfo: UserInfo): Route = {
```

Comment:
I would call this handleCreateGroup to be consistent with the others

Reply:
fixed